db/mongo.go

```diff
@@ -194,19 +200,73 @@ func (db *database) buildRestoreArgs(dumpPath string, includeStatusHistory bool)
 
 // RestoreFromDump uses mongorestore to load the dump from a backup.
 func (db *database) RestoreFromDump(dumpDir, logFile string, includeStatusHistory bool) error {
+	binary, isSnap, err := db.getRestoreBinary()
+	if err != nil {
+		return errors.Trace(err)
+	}
+
+	// Snap mongorestore can only access certain directories, so move the dump
+	// from /tmp to under $HOME/snap before running restore, and delete after.
+	if isSnap {
+		dumpDir, err = db.moveToHomeSnap(dumpDir)
```

Comment:
This should be a copy then a os.RemoveAll *only* on the copy. The original backup should stay behind.

Reply:
Yeah, that's what I was going to do at first, but I don't think it matters, and it's significantly cheaper to do this. The reason is that `juju-restore` has unzipped the tar.gz and deletes the unzipped contents before it's done.

I admit it's a little odd / non-symmetrical to delete the dump early and delete the rest later, but it feels bad to do a bunch of extra copying for this. Alternatively I guess I could *move* it back after restoring (in case the caller wants to access it). But seems quite hypothetical as the dump dir is only for `mongorestore`.

Reply:
Oh right, I didn't realize juju-restore unpacked the backup tar.